src/common/procedure-test/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Test utilities for procedures.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common_procedure::{
+    BoxedProcedure, Context, ContextProvider, ProcedureId, ProcedureState, Result, Status,
+};
+
+/// A Mock [ContextProvider] that always return [ProcedureState::Done].
+struct MockContextProvider {}
+
+#[async_trait]
+impl ContextProvider for MockContextProvider {
+    async fn procedure_state(&self, _procedure_id: ProcedureId) -> Result<Option<ProcedureState>> {
+        Ok(Some(ProcedureState::Done))
+    }
+}
+
+/// Executes a procedure until it returns [Status::Done].
+///
+/// # Panics
+/// Panics if the `procedure` has subprocedure to execute.
+pub async fn execute_procedure_until_done(procedure: &mut BoxedProcedure) {
```

Comment:
Can we reuse the `Runner::execute_procedure_in_loop`?

Reply:
The runner relies on the `ManagerContext`. I want a lightweight helper function to test these simple procedures. We still need such a utility crate so we can add a function to call execute N times in the future. Then the procedure can reach a specific state.